src/components/analyses/listing/Listing.js

```diff
@@ -141,40 +141,40 @@ function Listing(props) {
-                        ? message.payload.status
-                        : "";
-
-                const found = data?.analyses?.find(
-                    (analysis) => analysis.id === message.payload.id
-                );
-
-                if (found) {
-                    if (analysisStatus !== found.status) {
-                        found.status = analysisStatus;
-                        found.enddate = message.payload.enddate;
-                        setData({ analyses: [...data?.analyses] });
-                    }
-                } else {
-                    //add a new analysis record and remove the last record from the page
-                    //to maintain page size
-                    if (data?.analyses.length === rowsPerPage) {
-                        const newPage = data.analyses.slice(
-                            0,
-                            data.analyses.length - 1
-                        );
-                        setData({
-                            analyses: [message.payload, ...newPage],
-                        });
-                    } else if (data?.analyses.length === 0) {
-                        //if page is empty...
-                        setData({ analyses: [message.payload] });
+                ) {
+                    const analysisStatus = message.payload.status;
+                    const found = data?.analyses?.find(
+                        (analysis) => analysis.id === message.payload.id
+                    );
+
+                    if (found) {
+                        if (analysisStatus !== found.status) {
+                            found.status = analysisStatus;
+                            found.enddate = message.payload.enddate;
+                            setData({ analyses: [...data?.analyses] });
+                        }
                     } else {
-                        setData({
-                            analyses: [message.payload, ...data?.analyses],
-                        });
+                        //add a new analysis record and remove the last record from the page
+                        //to maintain page size
+                        if (data?.analyses.length === rowsPerPage) {
+                            const newPage = data.analyses.slice(
+                                0,
+                                data.analyses.length - 1
+                            );
+                            setData({
+                                analyses: [message.payload, ...newPage],
+                            });
+                        } else if (data?.analyses.length === 0) {
+                            //if page is empty...
+                            setData({ analyses: [message.payload] });
+                        } else {
+                            setData({
+                                analyses: [message.payload, ...data?.analyses],
+                            });
+                        }
```

Comment:
I think this can still throw an error.

When this listing is initially rendered, the `data` state is `null`, then the listing is fetched with a query; but if an analysis notification arrives while the listing is still loading, then `data` will still be `null` here, and the "uniterable" error will be thrown.

Another `if` check might prevent it (here or above when checking `category`):
```suggestion
                        } else if (data) {
                            setData({
                                analyses: [message.payload, ...data?.analyses],
                            });
                        }
```

Reply:
🤦 I had this change but removed to test one last time but I never put it back and then got distracted by data view issues. I should go to sleep!

Reply:
But that why we have code reviews 👍 (To save people like me.)